vpr/src/place/place.cpp

```diff
@@ -1796,45 +2123,78 @@ static void comp_td_connection_delays(const PlaceDelayModel* delay_model) {
+ * the placement quench. If successful, the setup slacks in PlacerSetupSlacks should be
+ * the same as the values in `connection_setup_slack` without running commit_setup_slacks().
+ * For more detailed info, see the try_swap() routine.
+ */
+static bool verify_connection_setup_slacks(const PlacerSetupSlacks* setup_slacks) {
+    const auto& clb_nlist = g_vpr_ctx.clustering().clb_nlist;
+
+    //Go through every single sink pin to check that the slack values are the same
+    for (ClusterNetId net_id : clb_nlist.nets()) {
+        for (size_t ipin = 1; ipin < clb_nlist.net_pins(net_id).size(); ++ipin) {
+            if (connection_setup_slack[net_id][ipin] != setup_slacks->setup_slack(net_id, ipin)) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
 /* Update the connection_timing_cost values from the temporary *
- * values for all connections that have changed.                   */
+ * values for all connections that have changed.               */
 static void commit_td_cost(const t_pl_blocks_to_be_moved& blocks_affected) {
     auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& clb_nlist = cluster_ctx.clb_nlist;
 
-    /* Go through all the blocks moved. */
-    for (int iblk = 0; iblk < blocks_affected.num_moved_blocks; iblk++) {
-        ClusterBlockId bnum = blocks_affected.moved_blocks[iblk].block_num;
-        for (ClusterPinId pin_id : cluster_ctx.clb_nlist.block_pins(bnum)) {
-            ClusterNetId net_id = cluster_ctx.clb_nlist.pin_net(pin_id);
-
-            if (cluster_ctx.clb_nlist.net_is_ignored(net_id))
-                continue;
-
-            if (cluster_ctx.clb_nlist.pin_type(pin_id) == PinType::DRIVER) {
-                //This net is being driven by a moved block, recompute
-                //all point to point connections on this net.
-                for (size_t ipin = 1; ipin < cluster_ctx.clb_nlist.net_pins(net_id).size(); ipin++) {
-                    connection_delay[net_id][ipin] = proposed_connection_delay[net_id][ipin];
-                    proposed_connection_delay[net_id][ipin] = INVALID_DELAY;
-                    connection_timing_cost[net_id][ipin] = proposed_connection_timing_cost[net_id][ipin];
-                    proposed_connection_timing_cost[net_id][ipin] = INVALID_DELAY;
-                }
-            } else {
-                //This pin is a net sink on a moved block
-                VTR_ASSERT_SAFE(cluster_ctx.clb_nlist.pin_type(pin_id) == PinType::SINK);
-
-                /* The following "if" prevents the value from being updated twice. */
-                if (!driven_by_moved_block(net_id, blocks_affected)) {
-                    int net_pin = cluster_ctx.clb_nlist.pin_net_index(pin_id);
+    //Go through all the sink pins affected
+    for (ClusterPinId pin_id : blocks_affected.affected_pins) {
+        ClusterNetId net_id = clb_nlist.pin_net(pin_id);
+        int ipin = clb_nlist.pin_net_index(pin_id);
```

Comment:
This is one of the few code differences in the base (CRITICALITY_PLACE) algorithm in the new code vs. the old code.  Code looks fine (cleaner), and affected_pins is pretty much the same pin list the other code would create dynamically.  One difference:  the old code did not update the timing cost if net_is_ignored(net_id) -- you could try putting that back in.  Doesn't seem like it would make much difference but may be worth a try.

Reply:
There's no difference due to the way ```affected_pins``` are gathered. 
This structure is populated by calling ```try_swap()```->```find_affected_nets_and_update_costs()```->```update_td_delta_costs()```.
The ignored nets are skipped in ```find_affected_nets_and_update_costs()```.